### PR TITLE
planner: add more metrics for binding

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #51347

Problem Summary: planner: add more metrics for binding

### What changed and how does it work?

image

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/metrics/bindinfo.go [hunk 1]
```diff
@@ -18,16 +18,48 @@ import "github.com/prometheus/client_golang/prometheus"
 
 // bindinfo metrics.
 var (
-	BindUsageCounter *prometheus.CounterVec
```
Comment (qw4990) [inline pkg/metrics/bindinfo.go:21]: Useless, just remove it

File: pkg/metrics/bindinfo.go [hunk 2]
```diff
@@ -18,16 +18,48 @@ import "github.com/prometheus/client_golang/prometheus"
...
-		}, []string{LabelScope})
+			Subsystem: "server",
+			Name:      "binding_cache_hit_total",
+			Help:      "Counter of binding cache hit.",
+		})
+	BindingCacheMissCounter = NewCounter(
+		prometheus.CounterOpts{
+			Namespace: "tidb",
+			Subsystem: "server",
+			Name:      "binding_cache_miss_total",
```
Comment (hawkingrei) [inline pkg/metrics/bindinfo.go:41]:
```suggestion
            Subsystem: "binding",
			Name:      "cache_miss_total",
```

Comment (hawkingrei) [reply pkg/metrics/bindinfo.go:41]: subsystem should be ```binding```. name can remove the ```binding``` string.

Comment (qw4990) [reply pkg/metrics/bindinfo.go:41]: Want to keep it in the `server` subsystem, same as Plan Cache.

Comment (hawkingrei) [reply pkg/metrics/bindinfo.go:41]: OK

**Review by hawkingrei — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Review by hi-rustin — APPROVED** [review]

**Pull request merged**